### PR TITLE
refactor: port segments list to react-table

### PR DESCRIPTION
**Pull request by olav**

### CODE REVIEW

**Review by NicolaeUnleash — CHANGES_REQUESTED** [review]
@olav 
Please can we have these adjustments:
1. Center the Actions column header text with the icons bellow
2. By default we should have a sorting criteria applied -> we could used the Created column
3. The search should highlight the words -> take a look on context fields page
4. When no results on search we should have -> There are no results for “keyword” <- (this should be a general solution in all the tables)
5. for the icon on the left side maybe we can use this one "DonutLarge" so it's not the same as on the context fields :D
6. Can you also update the label for the button like this "New segment" (segment with no capital letter)

**Comment by Tymek** [comment]
I'll adjust it to get this done today :)

> Please can we have these adjustments:

  - [x] 1. Center the Actions column header text with the icons bellow
  - [ ] 2. By default we should have a sorting criteria applied -> we could used the Created column
  - [ ] 3. The search should highlight the words -> take a look on context fields page
  - [ ] 4. When no results on search we should have -> There are no results for “keyword” <- (this should be a general solution in all the tables)
  - [ ] 5. for the icon on the left side maybe we can use this one "DonutLarge" so it's not the same as on the context fields :D
  - [ ] 6. Can you also update the label for the button like this "New segment" (segment with no capital letter)

**Review by NicolaeUnleash — APPROVED** [review]
this looks perfect now!

File: src/component/common/Table/Table/Table.styles.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}>()((theme, { rowHeight }) => ({
+    table: {
+        '& tbody tr': {
+            height:
+                {
+                    auto: 'auto',
+                    standard: theme.shape.tableRowHeight,
+                    compact: theme.shape.tableRowHeightCompact,
+                    dense: theme.shape.tableRowHeightDense,
+                }[rowHeight] || rowHeight,
```
Comment (nunogois) [inline src/component/common/Table/Table/Table.styles.ts:14]:
```suggestion
                }[rowHeight] ?? rowHeight,
```

File: src/component/common/Table/cells/HighlightCell/HighlightCell.tsx [hunk 2]
```diff
@@ -1,29 +1,42 @@
 import { VFC } from 'react';
-import { Box } from '@mui/material';
 import { Highlighter } from 'component/common/Highlighter/Highlighter';
-import { useSearchHighlightContext } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+import { useSearchHighlightContext } from '../../SearchHighlightContext/SearchHighlightContext';
```
Comment (nunogois) [inline src/component/common/Table/cells/HighlightCell/HighlightCell.tsx:3]:
```suggestion
import { useSearchHighlightContext } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
```

File: src/component/common/Table/cells/HighlightCell/HighlightCell.tsx [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+                className={classes.title}
+                style={{
+                    WebkitLineClamp: Boolean(subtitle) ? 1 : 2,
+                    lineClamp: Boolean(subtitle) ? 1 : 2,
+                }}
+                data-loading
+            >
+                <Highlighter search={searchQuery}>{value}</Highlighter>
             </span>
+            {Boolean(subtitle) && (
```
Comment (nunogois) [inline src/component/common/Table/cells/HighlightCell/HighlightCell.tsx:31]: Good candidate for `ConditionallyRender` instead?

File: src/component/common/Table/cells/HighlightCell/HighlightCell.tsx [hunk 4]
```diff
@@ -1,29 +1,42 @@
 import { VFC } from 'react';
-import { Box } from '@mui/material';
 import { Highlighter } from 'component/common/Highlighter/Highlighter';
-import { useSearchHighlightContext } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+import { useSearchHighlightContext } from '../../SearchHighlightContext/SearchHighlightContext';
+import { Box, Typography } from '@mui/material';
+import { useStyles } from '../HighlightCell/HighlightCell.styles';
```
Comment (nunogois) [inline src/component/common/Table/cells/HighlightCell/HighlightCell.tsx:5]:
```suggestion
import { useStyles } from './HighlightCell.styles';
```

File: src/component/project/ProjectAccess/ProjectAccessTable/ProjectRoleCell/ProjectRoleCell.styles.tsx [hunk 5]
```diff
@@ -2,6 +2,8 @@ import { makeStyles } from 'tss-react/mui';
 
 export const useStyles = makeStyles()(theme => ({
     cell: {
-        padding: theme.spacing(1, 1.5),
+        padding: theme.spacing(0, 1.5),
```
Comment (nunogois) [inline src/component/project/ProjectAccess/ProjectAccessTable/ProjectRoleCell/ProjectRoleCell.styles.tsx:5]: Huh, cool trick, did not know about https://mui.com/material-ui/customization/spacing/#multiple-arity 👍

File: src/component/segments/SegmentEmpty/SegmentEmpty.styles.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        margin: theme.spacing(6),
+        marginLeft: 'auto',
+        marginRight: 'auto',
+    },
+    title: {
+        fontSize: theme.fontSizes.mainHeader,
+        marginBottom: '1.25rem',
```
Comment (nunogois) [inline src/component/segments/SegmentEmpty/SegmentEmpty.styles.ts:15]:
```suggestion
        marginBottom: theme.spacing(2.5),
```

File: src/hooks/api/getters/useSegments/useSegments.ts [hunk 7]
```diff
@@ -18,7 +18,10 @@ export const useSegments = (strategyId?: string): IUseSegmentsOutput => {
 
     const { data, error, mutate } = useSWR(
         [strategyId, uiConfig.flags],
-        fetchSegments
+        fetchSegments,
+        {
+            refreshInterval: 15 * 1000,
```
Comment (nunogois) [inline src/hooks/api/getters/useSegments/useSegments.ts:23]: Curious: Why was this needed?

File: src/themes/theme.ts [hunk 8]
```diff
@@ -45,6 +45,7 @@ export default createTheme({
         borderRadiusLarge: '12px',
         borderRadiusExtraLarge: '20px',
         tableRowHeight: 64,
+        tableRowHeightCompact: 56,
```
Comment (nunogois) [inline src/themes/theme.ts:48]: I'm assuming these are 'px'. Would nice to keep the consistency when setting these values.

**Review by nunogois — CHANGES_REQUESTED** [review]
Overall looks good. Just some small suggestions.

**Comment by nunogois** [comment]
One small thing I forgot from my PR that you might want to add here for cleanup: Deleting `component/admin/projectRoles/ProjectRoles/ProjectRoleList/ProjectRoleListItem`. I believe it's not referenced anywhere else. Thanks 🙂

File: src/themes/theme.ts (hunk 8 above)
Comment (Tymek) [reply src/themes/theme.ts:48]: It's is used to calculate things, so I need `number`

File: src/hooks/api/getters/useSegments/useSegments.ts (hunk 7 above)
Comment (Tymek) [reply src/hooks/api/getters/useSegments/useSegments.ts:23]: Not strictly needed. It updates this list of other people add/remove items. Might be nice to add this as default for SWR.

**Review by nunogois — APPROVED** [review]
LGTM
